server/services/healthCheck.ts

```diff
@@ -0,0 +1,75 @@
+import { serviceCheckFactory } from '../data/healthCheck'
+import config from '../config'
+import type { AgentConfig } from '../config'
+
+interface HealthCheckStatus {
+  name: string
+  status: string
+  message: unknown
+}
+
+interface HealthCheckResult extends Record<string, unknown> {
+  healthy: boolean
+  checks: Record<string, unknown>
+}
+
+export type HealthCheckService = () => Promise<HealthCheckStatus>
+type HealthCheckCallback = (result: HealthCheckResult) => void
+
+function service(name: string, url: string, agentConfig: AgentConfig): HealthCheckService {
+  const check = serviceCheckFactory(name, url, agentConfig)
+  return () =>
+    check()
+      .then(result => ({ name, status: 'ok', message: result }))
+      .catch(err => ({ name, status: 'ERROR', message: err }))
+}
+
+function addAppInfo(result: HealthCheckResult): HealthCheckResult {
+  const buildInformation = getBuild()
+  const buildInfo = {
+    uptime: process.uptime(),
+    build: buildInformation,
+    version: buildInformation && buildInformation.buildNumber,
+  }
+
+  return { ...result, ...buildInfo }
+}
+
+function getBuild() {
+  try {
+    // eslint-disable-next-line import/no-unresolved,global-require
+    return require('../../build-info.json')
+  } catch (ex) {
+    return null
+  }
+}
+
+function gatherCheckInfo(aggregateStatus, currentStatus: HealthCheckStatus) {
+  return { ...aggregateStatus, [currentStatus.name]: currentStatus.message }
+}
+
+const apiChecks = [
+  service('hmppsAuth', `${config.apis.hmppsAuth.url}/health/ping`, config.apis.hmppsAuth.agent),
+  ...(config.apis.tokenVerification.enabled
+    ? [
+        service(
+          'tokenVerification',
+          `${config.apis.tokenVerification.url}/health/ping`,
+          config.apis.tokenVerification.agent
+        ),
+      ]
+    : []),
+]
+
+export default function healthCheck(callback: HealthCheckCallback, checks = apiChecks): void {
```

Comment:
wonder whether we could make this a bit nicer if the api didn't rely on needing a callback? 🤷‍♂️